### ISSUE TITLE
Shorthand not available in Swift

### ISSUE THREAD

**Issue by Isuru-Nanayakkara** [magicalpanda/MagicalRecord#936]
Hi,

I wanted to use MagicalRecord in a Swift project. I came across a few quirks. Let me explain.
##### Scenario No. 1

Following [this tutorial](http://jamesleist.com/core-data-via-magicalrecord-swift/), I integrated MagicalRecord to my Swift project using CocoaPods. 

```
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '8.0'
pod 'MagicalRecord/Shorthand'
```

Now note that this was written way before CocoaPods officially supported Swift. Therefore MagicalRecord is added as a static library which was the old way.

![3 old cocoapod first](https://cloud.githubusercontent.com/assets/2551877/5567473/64261246-8f6b-11e4-8f91-53852b2cec3f.png)

So to refer the method in the library, I had to create a Bridging header and all that jazz.

```
#define MR_SHORTHAND
#import <CoreData+MagicalRecord.h>
```

After that I could use MagicalRecord shorthand methods without any problem. I didn't have to import MagicalRecord in any files either.

```
let person = Person.createEntity() as Person
person.firstName = "Isuru"
person.lastName = "Nanayakkara"

NSManagedObjectContext.defaultContext().saveOnlySelfAndWait()
```
##### Scenario No. 2

As you may have heard, CocoaPods recently started [supporting](http://blog.cocoapods.org/) Swift.
So I recreated the above project. Note the `use_frameworks!` flag in the Podfile below.

```
source 'https://github.com/CocoaPods/Specs.git'
use_frameworks!
platform :ios, '8.0'
pod 'MagicalRecord/Shorthand'
```

Now the library is added as a framework. As it should for Swift projects.
![screen shot 2014-12-27 at 9 28 33 pm](https://cloud.githubusercontent.com/assets/2551877/5567532/8f87a336-8f6c-11e4-8091-5f07e67b2338.png)

Here comes the problems.
In the AppDelegate, I had to explicitly `import MagicalRecord` to call the `setupCoreDataStack()` method. No big deal.
In a view controller, I tried to write the same code snippet for creating an instance of a core data object but I couldn't reference any of the shorthand methods! But the methods with `MR_` prefix were available. I also explicitly had to import CoreData framework to the file because it wouldn't recognize the `NSManagedObjectContext` class without it.

```
let person = Person.MR_createEntity() as Person
person.firstName = "Isuru"
person.lastName = "Nanayakkara"

NSManagedObjectContext.MR_defaultContext().MR_saveOnlySelfAndWait()
```

I uploaded two Xcode projects demonstrating the two scenarios here.
[Scenario 1](https://www.dropbox.com/s/zk1z45u3kz3wifx/Magic-old.zip?dl=0)
[Scenario 2](https://www.dropbox.com/s/serj1c9y48wmabo/Magic-swiftpod.zip?dl=0)
In both projects, the core data classes are generated using [Mogenerator](https://github.com/rentzsch/mogenerator).

Any idea why this is happening? Is this a bug or am I missing something?

**Comment by tonyarnold** [comment]
No, it's just not something we've implemented yet. MagicalRecord's shorthand was implemented using method swizzling — I've no idea how that plays with Swift. 

What we'll likely have to do is create a Swift "shim" like projects such as Realm and ReactiveCocoa have done.

**Comment by tonyarnold** [comment]
You might be interested in trying out the work in #960 — it enables shorthand by default, and includes a proper dynamic framework for OS X and iOS.

**Comment by Isuru-Nanayakkara** [comment]
Thank you, Tony. Will keep a lookout on that. :+1:
